Mathlib/Analysis/Fourier/Inversion.lean

```diff
@@ -0,0 +1,159 @@
+      GaussianFourier.integrable_cexp_neg_mul_sq_norm_add (by simpa) _ _
+    simpa using (VectorFourier.integral_fourierIntegral_smul_eq_flip (L := innerₗ V)
+      Real.continuous_fourierChar continuous_inner I hf).symm
+  apply B.congr'
+  filter_upwards [Ioi_mem_atTop 0] with c (hc : 0 < c)
+  congr with w
+  rw [fourierIntegral_gaussian_innerProductSpace' (by simpa)]
+  congr
+  · simp
+  · simp; ring
+
+lemma tendsto_integral_gaussian_smul' (hf : Integrable f) {v : V} (h'f : ContinuousAt f v) :
+    Tendsto (fun (c : ℝ) ↦
+      ∫ w : V, ((π * c : ℂ) ^ (finrank ℝ V / 2 : ℂ) * cexp (-π ^ 2 * c * ‖v - w‖ ^ 2)) • f w)
+    atTop (𝓝 (f v)) := by
+  let φ : V → ℝ := fun w ↦ π ^ (finrank ℝ V / 2 : ℝ) * Real.exp (-π^2 * ‖w‖^2)
+  have A : Tendsto (fun (c : ℝ) ↦ ∫ w : V, (c ^ finrank ℝ V * φ (c • (v - w))) • f w)
+      atTop (𝓝 (f v)) := by
+    apply tendsto_integral_comp_smul_smul_of_integrable'
+    · exact fun x ↦ by positivity
+    · rw [integral_mul_left, GaussianFourier.integral_rexp_neg_mul_sq_norm (by positivity)]
+      nth_rewrite 2 [← pow_one π]
+      rw [← rpow_nat_cast, ← rpow_nat_cast, ← rpow_sub pi_pos, ← rpow_mul pi_nonneg,
+        ← rpow_add pi_pos]
+      ring_nf
+      exact rpow_zero _
+    · have A : Tendsto (fun (w : V) ↦ π^2 * ‖w‖^2) (cobounded V) atTop := by
+        rw [tendsto_const_mul_atTop_of_pos (by positivity)]
+        apply (tendsto_pow_atTop two_ne_zero).comp tendsto_norm_cobounded_atTop
+      have B := tendsto_rpow_mul_exp_neg_mul_atTop_nhds_zero (finrank ℝ V / 2) 1
+        zero_lt_one |>.comp A |>.const_mul (π ^ (-finrank ℝ V / 2 : ℝ))
+      rw [mul_zero] at B
+      convert B using 2 with x
+      simp only [neg_mul, one_mul, Function.comp_apply, ← mul_assoc, ← rpow_nat_cast, φ]
+      congr 1
+      rw [mul_rpow (by positivity) (by positivity), ← rpow_mul pi_nonneg,
+        ← rpow_mul (norm_nonneg _), ← mul_assoc, ← rpow_add pi_pos, mul_comm]
+      congr <;> ring
+    · exact hf
+    · exact h'f
+  have B : Tendsto
+      (fun (c : ℝ) ↦ ∫ w : V, ((c^(1/2:ℝ)) ^ finrank ℝ V * φ ((c^(1/2:ℝ)) • (v - w))) • f w)
+      atTop (𝓝 (f v)) :=
+    A.comp (tendsto_rpow_atTop (by norm_num))
+  apply B.congr'
+  filter_upwards [Ioi_mem_atTop 0] with c (hc : 0 < c)
+  congr with w
+  rw [← coe_smul]
+  congr
+  rw [ofReal_mul, ofReal_mul, ofReal_exp, ← mul_assoc]
+  congr
+  · rw [mul_cpow_ofReal_nonneg pi_nonneg hc.le, ← rpow_nat_cast, ← rpow_mul hc.le, mul_comm,
+      ofReal_cpow pi_nonneg, ofReal_cpow hc.le]
+    simp [div_eq_inv_mul]
+  · norm_cast
+    simp only [one_div, norm_smul, Real.norm_eq_abs, mul_pow, _root_.sq_abs, neg_mul, neg_inj,
+      ← rpow_nat_cast, ← rpow_mul hc.le, mul_assoc]
+    norm_num
+
+/-- **Fourier inversion formula**: If a function `f` on a finite-dimensional real inner product
+space is integrable, and its Fourier transform `𝓕 f` is also integrable, then `𝓕⁻ (𝓕 f) = f` at
+continuity points of `f`. -/
+theorem fourier_inversion (hf : Integrable f) (h'f : Integrable (𝓕 f)) {v : V}
```

Comment:
Does it make sense to state a version for continuous `f` that gives equality of functions? (I.e., after `funext`, apply this current version.)